src/main/java/nl/tudelft/pixelperfect/client/ServerListener.java

```diff
@@ -52,6 +63,8 @@ public void messageReceived(HostedConnection source, Message message) {
       EventCompletedMessage eve = (EventCompletedMessage) message;
       System.out.println("Received a completed event: " + eve.getLabel());
       app.getSpaceship().getLog().complete(eve.getCompletedEvent());
+    } else if (message instanceof RoleChosenMessage) {
+      server.broadcast(message);
     }
```

Comment:
I'm honestly not quite sure that this is supposed to do.
Is it supposed to send a message to everyone else that a player chose a role?
Because if that's the case, you could try and set a filter as parameter for the broadcast method to exclude the client that sent the message.